### ISSUE TITLE
Protected properties cause TypeInitializationException 

### ISSUE THREAD

**Issue by niklasbjorkman** [Starcounter/Starcounter#3505]
If you have a non-persistent property that are declared protected you get a TypeInitializationException when you try to instance the class. No further information is given on why there is a problem.

I think you should be able to declare protected properties in your persistent class. A nice message on WHY you cannot initialize the class would be much appreciated. I have not tried declare protected fields, so those might be problem as well?

Our example
public class Pallet
{
     protected String QuantityDescription
     {
             get{ ....}
      }
}

Chage the property to public and you can run it again.

**Comment by per-samuelsson** [comment]
Thanks for reporting this.

Unfortunately, issues with non-public fields and properties is something we've been aware of for quite some time. Here's a set of related issues: #2519, #1614, #2520, #2707.

It's one of those things we clearly should fix properly, but never get a timeslot to do so. The biggest obstacle is still that we don't know how those language-specific access modifiers should relate to our SQL implementation. It's not easily mapped 1-1. That's still is no excuse we can't make an effort to provide better error information of course.

**Comment by niklasbjorkman** [comment]
FYI. Took us 1 hours times 2 persons to figure out why it constantly crashed (a structure of 4 inherited classes with a lot of fields). So a proper error message is definitely urgent.

Our opinion is that the NonPersistent fields should be able to define private/protected to support good OO design. For us it is better to skip support on SQL on top of these properties. Could be dangerous anyway if we have a property somewhere, for instance, that fetch all receipt items for a specific day. That should not be invoked during querying ...

**Comment by k-rus** [comment]
I agree that non-persistent private/protected properties can be just ignored, but then they should be explicitly defined to be non-persistent.

**Comment by per-samuelsson** [comment]
> Took us 1+ hour times 2 persons to figure out why it constantly crashed (a structure of 4 inherited classes with a lot of fields). So a proper error message is definitely urgent.

Yes, clearly, totally agree. Sorry you had to experience that.
